[addons/storage-provisioner-gluster] Use gluster's official container…

black-dragon74 opened the pull request:
… images

I have omitted the sha256 sum as we only ever provide the `latest`(aka `centos`) and `fedora`
tags. We don't have versioned releases. Including the SHA sum would require us to update the sum
here on every new release which would be a hassle.

Signed-off-by: black.dragon74 <nickk.2974@gmail.com>

<!-- 🎉 Thank you for contributing to minikube! 🎉 Here are some hints to get your PR merged faster:

1. Your PR title will be included in the release notes, choose it carefully
2. If the PR fixes an issue, add "fixes #<issue number>" to the description.
3. If the PR is a user interface change, please include a "before" and "after" example.
4. If the PR is a large design change, please include an enhancement proposal:
https://github.com/kubernetes/minikube/tree/master/enhancements
-->

black-dragon74 commented:
/cc: @nixpanic @spowelljr

RA489 commented:
/easycla

spowelljr commented:
Hi @black-dragon74, you'll need to sign the CLA as described in https://github.com/kubernetes/minikube/pull/14609#issuecomment-1188819110 for us to proceed with the PR.

Not having the SHA for the image is a security problem for us, if someone were able to get the credentials to your Docker Hub repo they could push up a malicious image and any user that went to use the image would be compromised.

Is there a reason you don't keep previous images up? I think it would be advantageous to more people than just us if previous images were available. Being forced to a new image without being able to scan or vet it before hand is a big security issue and not something we want to introduce into the code base.

black-dragon74 commented:
> Hi @black-dragon74, you'll need to sign the CLA as described in [#14609 (comment)](https://github.com/kubernetes/minikube/pull/14609#issuecomment-1188819110) for us to proceed with the PR.

Sure, will do.



> Being forced to a new image without being able to scan or vet it before hand is a big security issue and not something we want to introduce into the code base.

Agreed. I'll see what we can do about it. Let's keep this PR open until then. I'll ping you once we decide on something.

Regards

black-dragon74 commented:
@spowelljr We changed the way we publish releases and now we have a SHA sum that will be available for verification at all times. We no longer prune old tags. I've updated and rebased the PR for the same.

Regards

spowelljr commented:
Hi @black-dragon74, sorry, I didn't see your previous ping. I'll try this out and if it works I'll be happy to merge it, thanks

spowelljr commented:
@black-dragon74 I tried it locally on my machine and the new image causes a CrashLoopBackOff.

```
Name:         glusterfs-t59mc
Namespace:    storage-gluster
Priority:     0
Node:         minikube/192.168.58.2
Start Time:   Wed, 12 Oct 2022 15:51:33 -0700
Labels:       controller-revision-hash=7f95c9c749
              glusterfs=pod
              glusterfs-node=pod
              k8s-app=storage-provisioner-gluster
              pod-template-generation=1
Annotations:  <none>
Status:       Running
IP:           192.168.58.2
IPs:
  IP:           192.168.58.2
Controlled By:  DaemonSet/glusterfs
Containers:
  glusterfs:
    Container ID:   docker://9c7149822ebcbf44f637079dce5e30670a40767eff47d4d0c9773e805a947ac1
    Image:          ghcr.io/gluster/gluster-containers:glfs10.2-centos@sha256:961c1d8169ec61831c7b0650f9d099460581a46b2778202590f438ea7f857e21
    Image ID:       docker-pullable://ghcr.io/gluster/gluster-containers@sha256:961c1d8169ec61831c7b0650f9d099460581a46b2778202590f438ea7f857e21
    Port:           <none>
    Host Port:      <none>
    State:          Waiting
      Reason:       CrashLoopBackOff
    Last State:     Terminated
      Reason:       Error
      Exit Code:    255
      Started:      Wed, 12 Oct 2022 15:53:29 -0700
      Finished:     Wed, 12 Oct 2022 15:53:29 -0700
    Ready:          False
    Restart Count:  4
    Requests:
      cpu:      100m
      memory:   100Mi
    Liveness:   exec [/bin/bash -c systemctl status glusterd.service] delay=40s timeout=3s period=25s #success=1 #failure=50
    Readiness:  exec [/bin/bash -c systemctl status glusterd.service] delay=40s timeout=3s period=25s #success=1 #failure=50
    Environment:
      USE_FAKE_DISK:  enabled
    Mounts:
      /dev from glusterfs-dev (rw)
      /etc/ssl from glusterfs-ssl (ro)
      /run from glusterfs-run (rw)
      /run/lvm from glusterfs-lvm (rw)
      /srv from fake-disk (rw)
      /sys/fs/cgroup from glusterfs-cgroup (ro)
      /usr/lib/modules from kernel-modules (ro)
      /var/lib/glusterd from glusterfs-config (rw)
      /var/lib/heketi from glusterfs-heketi (rw)
      /var/lib/misc/glusterfsd from glusterfs-misc (rw)
      /var/log/glusterfs from glusterfs-logs (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from kube-api-access-x5pq6 (ro)
Conditions:
  Type              Status
  Initialized       True
  Ready             False
  ContainersReady   False
  PodScheduled      True
Volumes:
  fake-disk:
    Type:          HostPath (bare host directory volume)
    Path:          /srv
    HostPathType:
  glusterfs-heketi:
    Type:          HostPath (bare host directory volume)
    Path:          /var/lib/heketi
    HostPathType:
  glusterfs-run:
    Type:       EmptyDir (a temporary directory that shares a pod's lifetime)
    Medium:
    SizeLimit:  <unset>
  glusterfs-lvm:
    Type:          HostPath (bare host directory volume)
    Path:          /run/lvm
    HostPathType:
  glusterfs-etc:
    Type:          HostPath (bare host directory volume)
    Path:          /etc/glusterfs
    HostPathType:
  glusterfs-logs:
    Type:          HostPath (bare host directory volume)
    Path:          /var/log/glusterfs
    HostPathType:
  glusterfs-config:
    Type:          HostPath (bare host directory volume)
    Path:          /var/lib/glusterd
    HostPathType:
  glusterfs-dev:
    Type:          HostPath (bare host directory volume)
    Path:          /dev
    HostPathType:
  glusterfs-misc:
    Type:          HostPath (bare host directory volume)
    Path:          /var/lib/misc/glusterfsd
    HostPathType:
  glusterfs-cgroup:
    Type:          HostPath (bare host directory volume)
    Path:          /sys/fs/cgroup
    HostPathType:
  glusterfs-ssl:
    Type:          HostPath (bare host directory volume)
    Path:          /etc/ssl
    HostPathType:
  kernel-modules:
    Type:          HostPath (bare host directory volume)
    Path:          /usr/lib/modules
    HostPathType:
  kube-api-access-x5pq6:
    Type:                    Projected (a volume that contains injected data from multiple sources)
    TokenExpirationSeconds:  3607
    ConfigMapName:           kube-root-ca.crt
    ConfigMapOptional:       <nil>
    DownwardAPI:             true
QoS Class:                   Burstable
Node-Selectors:              <none>
Tolerations:                 node.kubernetes.io/disk-pressure:NoSchedule op=Exists
                             node.kubernetes.io/memory-pressure:NoSchedule op=Exists
                             node.kubernetes.io/network-unavailable:NoSchedule op=Exists
                             node.kubernetes.io/not-ready:NoExecute op=Exists
                             node.kubernetes.io/pid-pressure:NoSchedule op=Exists
                             node.kubernetes.io/unreachable:NoExecute op=Exists
                             node.kubernetes.io/unschedulable:NoSchedule op=Exists
Events:
  Type     Reason     Age                  From               Message
  ----     ------     ----                 ----               -------
  Normal   Scheduled  2m25s                default-scheduler  Successfully assigned storage-gluster/glusterfs-t59mc to minikube
  Normal   Pulling    2m24s                kubelet            Pulling image "ghcr.io/gluster/gluster-containers:glfs10.2-centos@sha256:961c1d8169ec61831c7b0650f9d099460581a46b2778202590f438ea7f857e21"
  Normal   Pulled     2m11s                kubelet            Successfully pulled image "ghcr.io/gluster/gluster-containers:glfs10.2-centos@sha256:961c1d8169ec61831c7b0650f9d099460581a46b2778202590f438ea7f857e21" in 13.215670509s
  Normal   Started    84s (x4 over 2m10s)  kubelet            Started container glusterfs
  Warning  BackOff    44s (x10 over 2m9s)  kubelet            Back-off restarting failed container
  Normal   Created    29s (x5 over 2m10s)  kubelet            Created container glusterfs
  Normal   Pulled     29s (x4 over 2m10s)  kubelet            Container image "ghcr.io/gluster/gluster-containers:glfs10.2-centos@sha256:961c1d8169ec61831c7b0650f9d099460581a46b2778202590f438ea7f857e21" already present on machine
```

```
Failed to connect to bus: No such file or directory
Systemd is using legacy cgroups(v1)
cat: /sys/fs/cgroup/pids//pids.max: No such file or directory
maximum number of pids configured in cgroups:
/usr/local/bin/update-params.sh: line 39: /sys/fs/cgroup/pids//pids.max: No such file or directory
cat: /sys/fs/cgroup/pids//pids.max: No such file or directory
maximum number of pids configured in cgroups (reconfigured):
env variable is set. Update in gluster-blockd.service
```

black-dragon74 commented:
> /sys/fs/cgroup from glusterfs-cgroup (ro)

Hi Steven, thanks for taking a look at it. Actually the options to run the container have changed.

- `cgroup` mount type needs to be `rw` as we use the host namespace for it. Hence the flag `--cgroupns=host` is also required.

P.S: Regarding the changes I can remove the `readOnly` from cgroup mount in `glusterfs-daemonset.yaml.tmpl`. Where can I add this `cgroupns` flag? Please help me with this and I'll send a PR right away.

Regards
